src/components/foodDescription/FoodDescription.js

```diff
@@ -1,44 +1,122 @@
 // React
-import { useMemo, useRef } from "react";
+import { useMemo, useRef, useState } from "react";
 
 // Libraries
+import React from "react";
 import PropTypes from "prop-types";
 
 // lodash
 import _keys from "lodash/keys";
 import _toLower from "lodash/toLower";
+import _has from "lodash/has";
+import _omit from "lodash/omit";
+import _isEmpty from "lodash/isEmpty";
 import _find from "lodash/find";
 
 // components
 import Sidebar from "./components/sidebar";
 import FoodList from "./components/foodList";
+import Cart from "../cart";
+
+// helpers
+import { getCartItems, getTotalFromCartItems } from "./helpers/foodDescription.general";
+
+// utility
+import saveCartItems from "../../utility/saveCartItems";
+
+// icons
+import emptyCartImage from '../../assets/images/cartEmpty.webp'
 
 // css
 import "./foodDescription.css";
 
 function FoodDescription(props) {
     const { foodList } = props;
+
     const keys = useMemo(() => _keys(foodList), [foodList]);
     const foodListSectionRefs = useRef([]);
 
+    const [cart, setCart] = useState({});
+
     function setRef(ele) {
         if (!foodListSectionRefs.current.includes(ele)) {
             foodListSectionRefs.current.push(ele);
         }
     }
 
     function handleCategoryScroll(category) {
-        const section = _find(foodListSectionRefs.current, { id: _toLower(category) });
+        const section = _find(foodListSectionRefs.current, {
+            id: _toLower(category),
+        });
         section.scrollIntoView({ behavior: "smooth" });
     }
 
+    function handleAddToCart(id) {
+        const isIdPresentInCart = _has(cart, id);
+        if (!isIdPresentInCart) {
+            setCart({ ...cart, [id]: 1 });
+        }
+    }
+
+    function handleIncrementClick(id) {
+        const newQuantitiy = cart[id] + 1;
+        setCart({ ...cart, [id]: newQuantitiy });
+    }
+
+    function handleDecrementClick(id) {
+        const newQuantitiy = cart[id] - 1;
+        if (newQuantitiy === 0) {
```

Comment:
handle this condition while rendering the UI

Reply:
i didnt get your point
if the quantity is 1 and decrement is done then i will remove the id and set the state